RefrigeratorSaraFarkash/RefrigeratorSaraFarkash/CRefrigerator.cs

```diff
@@ -0,0 +1,334 @@
+        }
+
+        public void AddShelfs(CShelf shelf)
+        {
+            if (shelfs.Count >= numberOfShelves)
+            {
+                Console.WriteLine("we cant add this shelfs, there is no place!!");
+            }
+            else
+            {
+                shelfs.Add(shelf);
+                Console.WriteLine("the shefs added!!");
+            }
+
+        }
+        public override string ToString()
+        {
+            StringBuilder result = new StringBuilder();
+            result.AppendLine($"Department ID: {Id}");
+            result.AppendLine($"Model: {Model}");
+            result.AppendLine($"Color: {Color}");
+            result.AppendLine($"Number of Shelves: {NumberOfShelves}");
+
+            // Print details for each shelf
+            result.AppendLine("Shelves:");
+
+            foreach (var shelf in Shelfs)
+            {
+                result.AppendLine(shelf.ToString());
+            }
+
+            return result.ToString();
+
+        }
+
+        //2
+        public double spaceLeftInRefrigerator()
+        {
+            double space = 0.0;
+            if (numberOfShelves >= 0)
+            {
+                for (int i = 0; i < shelfs.Count; i++)
+                {
+                    space = space + shelfs[i].spaceLeftInShelfs();
+                }
+            }
+            else
+            {
+                Console.WriteLine("Invalid number of shelves. Cannot calculate space.");
+            }
+
+            return space;
+        }
+
+        //3
+        public void pushToRefItem(CItem item1)
+        {
+
+            if (this.spaceLeftInRefrigerator() < item1.TakeSpace)
+                Console.WriteLine("there is no space!!!");
+            else
+            {
+                for (int i = 0; i < this.numberOfShelves; i++)
+                {
+                    if (this.shelfs[i].spaceLeftInShelfs() > item1.TakeSpace)
+                    {
+                        item1.IdShelf = this.shelfs[i].Id;
+                        this.shelfs[i].AddItem(item1);
+                        Console.WriteLine("the item is added!!!");
+                        break;
+                    }
+                }
+            }
+        }
+
+        //4
+        public CItem removeItemFromRef(int idItem)
+        {
+            CItem itemRemove;
+            for (int i = 0; i < this.numberOfShelves; i++)
+            {
+                if (shelfs[i].findItemInShelf(idItem))
+                {
+                    itemRemove = shelfs[i].removeItemfromShelf(idItem);
+                    return itemRemove;
+                }
+            }
+
+            Console.WriteLine("the item not found!!!");
+            return null;
+        }
+        //5
+        public void cleanExpired()
+        {
+            for (int i = 0; i < shelfs.Count; i++)
+            {
+                List<CItem> itemsToRemove = new List<CItem>();
+                foreach (CItem item in shelfs[i].Items)
```

Comment:
should have a cleanExpired on each shelf and call it so each shelf cleans itself